src/libraries/System.Formats.Tar/tests/TarFile/TarFile.ExtractToDirectory.Base.cs

```diff
@@ -0,0 +1,80 @@
+    {
+        var allValidFormats = new TarEntryFormat[] { TarEntryFormat.V7, TarEntryFormat.Ustar, TarEntryFormat.Pax, TarEntryFormat.Gnu };
+
+        foreach (TarEntryFormat format in allValidFormats)
+        {
+            yield return new object[]
+            {
+                    format,
+                    TarEntryType.Directory,
+                    "" // Root directory
+            };
+            yield return new object[]
+            {
+                    format,
+                    TarEntryType.Directory,
+                    "./" // Slash dot root directory
+            };
+            yield return new object[]
+            {
+                    format,
+                    TarEntryType.Directory,
+                    "directory",
+            };
+            yield return new object[]
+            {
+                    format,
+                    GetTarEntryTypeForTarEntryFormat(TarEntryType.RegularFile, format),
+                    "file.txt"
+            };
+        }
+
+        var formatsThatHandleLongFileNames = new TarEntryFormat[] { TarEntryFormat.Pax, TarEntryFormat.Gnu };
+        var longFileNames = new string[]
+        {
+            // Long path with many short segment names  and a filename
+            "folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/file.txt",
+            // Long path with single long segment name and a filename
+            "veryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryverylongfoldername/file.txt",
+            // Long path with single long leaf filename
+            "veryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryverylongfilename.txt",
+        };
+
+        foreach (TarEntryFormat format in formatsThatHandleLongFileNames)
+        {
+            foreach (string filePath in longFileNames)
+            {
+                yield return new object[] { format, TarEntryType.RegularFile, filePath };
+            }
+        }
+
+        var longFolderNames = new string[]
+        {
+            // Long path with many short segment names  and a filename
+            "folderfolder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder/folder",
```

Comment:
Can you generate these rather than paste in big strings? It's self documenting. Tests do this various ways eg string.Create, Enumerable.Repeat, maybe there's some Utility class. The micro benchmarks have another way, etc